corehq/form_processor/models.py

```diff
@@ -892,7 +892,8 @@ def __unicode__(self):
 
     class Meta:
         index_together = [
-            ["domain", "owner_id", "closed"],
+            ["owner_id", "closed"],  # pghero recommended just indexing owner_id, but that's probably because there's few closed cases
+            ["owner_id", "server_modified_on"],
```

Comment:
Not having domain in the query would definitely be a vulnerability, but that would be in our query code, not the index.

I think that the reason it suggested this is that you want the first part of the index to drill down as much of the rows as possible. There will always be owner_id >> domain no matter the environment which means that when searching on a query that includes both domain and owner_id, owner_id is 
 the much more useful filter. 

I think the difference is:

with [domain, owner_id, closed] the index says, get me the records that match this domain, inside that domain get me the records from this owner_id.

with [owner_id, closed] the index says get me the records that match this owner id, then i'll check those records to make sure they have that domain. Because  owner_id should always be in only one domain (unless it's a web user) then this should keep the index smaller which I believe is more efficient. 

I obviously haven't been able to try this on a real production database, so I'm not sure how much this would improve it